08 - Fun with HTML5 Canvas/index-CLASS.html

```diff
@@ -0,0 +1,128 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+  <meta charset="UTF-8">
+  <title>HTML5 Canvas</title>
+</head>
+  <body>
+    <canvas id="draw" width="800" height="800"></canvas>
+    <button class="clearCanvas">Clear Canvas</button>
+    <script>
+        //// classes
+        class DrawingCanvas {
+            constructor(id, contextID = '2d') {
+                this.canvas = document.querySelector(id);
+                this.context = this.canvas.getContext(contextID)
+                this.X = 0;
+                this.Y = 0;
+                this.hue = 0;
+                this.isDrawing = false;
+                this.isStrokeGrowing = true;
+
+                //// binding this to events
+                this.startDrawing = this.startDrawing.bind(this);
+                this.stopDrawing = this.stopDrawing.bind(this);
+                this.clearCanvas = this.clearCanvas.bind(this);
+                this.draw = this.draw.bind(this);
+            }
+
+            setCanvas() {
+                this.canvas.width = window.innerWidth;
+                this.canvas.height = window.innerHeight;
+                this.context.strokeStyle = '#000';
+                this.context.lineJoin = 'miter';
+                this.context.lineCap = 'round';
+                this.context.lineWidth = 1;
+            }
+
+            updateCoordinates() {
+                this.X = this.offsetX;
+                this.Y = this.offsetY;
+            };
+
+            startDrawing() {
```

Comment:
nice that you put this as an explicit method. you could also have set the field/property directly in the event listener, but this looks nice.

Reply:
thx! I like calling methods. Feels like casting a spell or clicking on play/stop on the stereo.

Reply:
haha. harry potter says hi :)